terraform/modules/enclave/prometheus/cloud.conf

```diff
@@ -0,0 +1,53 @@
+#cloud-config
+apt_preserve_sources_list: true
+apt_sources:
+ - source: "deb [arch=amd64] http://www.mirrorservice.org/sites/archive.ubuntu.com/ubuntu bionic main restricted universe multiverse"
+ - source: "deb [arch=amd64] http://www.mirrorservice.org/sites/archive.ubuntu.com/ubuntu bionic-security main restricted universe multiverse"
+ - source: "deb [arch=amd64] http://www.mirrorservice.org/sites/archive.ubuntu.com/ubuntu bionic-updates main restricted universe multiverse"
+
+package_update: true
+package_upgrade: true
+packages: ['prometheus', 'prometheus-node-exporter', 'awscli', 'inotify-tools']
+
+write_files:
+  - owner: root:root
+    path: /etc/default/prometheus
+    permissions: 0444
+    content: 'ARGS="--storage.tsdb.path=\"/mnt/\""'
+  - owner: root:root
+    path: /etc/cron.d/config_pull
+    content: |
+        */2 * * * * root aws s3 sync s3://${config_bucket}/prometheus/ /etc/prometheus/ --region=${region}
+        @reboot root mount /dev/xvdh /mnt
+        @reboot root /root/watch_prometheus_dir
+  - content: |
+       #!/bin/bash
+       if file -s /dev/xvdh | grep -q "/dev/xvdh: data"; then
+         mkfs -t 'ext4' -L 'prometheus_disk' '/dev/xvdh'
+       else
+         echo "disk already formated"
+       fi
+    path: /root/format_disk.sh
+    permissions: 0755
+  - content: |
+       #!/bin/bash
+       inotifywait -e modify,create,delete,move -m -r /etc/prometheus |
+       while read -r directory events; do
+         systemctl reload prometheus
+       done
+    path: /root/watch_prometheus_dir
+    permissions: 0755
+  - content: |
+       Acquire::http::Proxy "http://${egress_proxy}";
+    path: '/etc/apt/apt.conf.d/05proxy'
+    permissions: '0644'
+  - content: ''
+    path: '/etc/apt/sources.list'
+
+runcmd:
+  - "if [ '${aws_ec2_ip}' ]; then echo '${aws_ec2_ip} ec2.eu-west-2.amazonaws.com' >> /etc/hosts; fi"
+  - [ bash, -c, "/root/format_disk.sh"]
+  - [ bash, -c, "mount /dev/xvdh /mnt"]
+  - [bash, -c, "chown -R prometheus /mnt/"]
+  - [systemctl, restart, prometheus]
```

Comment:
I think this was just to ensure that Prometheus starts successfully as we discussed with Ed. We will just stick to one reboot and start on a new state of the instance. This will do for the time being.